pages/best-practices/best-practices.mdx

```diff
@@ -0,0 +1,293 @@
+# Best Practices
+
+## Rule Categories
+
+### Priority A: Essential
+
+The essential rules are the most important ones.
+Use them to ensure that your SpiceDB cluster is performant, your schema is sane, and your authorization logic is sound.
+Exceptions to these rules should be rare and well justified.
+
+### Priority B: Strongly Recommended
+
+The strong recommendation rules will improve the schema design, developer experience, and performance of your SpiceDB cluster.
+In most cases, these rules should be followed.
+
+### Priority C: Recommended
+
+The recommended rules reflect how we would run our own systems, but may not apply to every use case and may not make sense in every situation.
+Follow them if you can and ignore them if you can’t.
+
+## Priority A Rules: Essential
+
+### Make sure your schema fails closed
+
+Tags: **schema**
+
+This is related to the idea of using negation sparingly, and of phrasing your schema additively.
+Give thought to what happens if your application fails to write a relation: should the user have access in that case?
+The answer is almost always `no`.
+
+### Tune Connections to Datastores
```

Comment:
Are there any other flags that should be tuned when configuring in production? For example: concurrency settings? cc @josephschorr 

Can we add them as a separate section, or maybe make this section more generic with "Tune Performance in Production", or maybe even create a separate page altogether?

Reply:
We talked about this and even brought in Victor to ask about other things to pay attention to for tuning.

Most of them were action items for things that could have better defaults: https://linear.app/authzed/project/offsite-2025-action-items-d78bcbecafb3/issues

And database connection tuning is the only thing that we found that varies meaningfully from deployment to deployment and also can't be automatically configured (at least in open-source).